### PR TITLE
feat: lesser restrict on multisig since

### PR DESCRIPTION
**Pull request by jjyr**
Multisig checks the since field of every input and makes sure:
1.  the flags of `input since` must be the same with the flags of `script since`.
2.  the value of `input since` must be greater or equals to value of`script since`.

### CODE REVIEW

File: c/common.h [hunk 1]
```diff
@@ -69,3 +76,90 @@ int extract_witness_lock(uint8_t *witness, uint64_t len,
   *lock_bytes_seg = MolReader_Bytes_raw_bytes(&lock_seg);
   return CKB_SUCCESS;
 }
+
+/* a and b are since value,
+ return 0 if a is equals to b,
+ return -1 if a is less than b,
+ return 1 if a is greater than b */
+int epoch_number_with_fraction_cmp(uint64_t a, uint64_t b) {
```
Comment (xxuejie) [inline c/common.h]: Can we bump this compare function to `ckb_syscalls.h` or a different utility file? I think it could be useful in many places.

Comment (jjyr) [reply c/common.h]: Sure, I think to put it in another `utils.h` is good, don't mix the utilities with syscalls.

File: c/utils.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  uint64_t b_index = (b >> INDEX_OFFSET) & INDEX_MASK;
+  uint64_t b_len = (b >> LENGTH_OFFSET) & LENGTH_MASK;
+
+  if (a_epoch < b_epoch) {
+    return -1;
+  } else if (a_epoch > b_epoch) {
+    return 1;
+  } else {
+    /* a and b is in the same epoch */
+    uint64_t a_block = a_index * a_len;
```
Comment (doitian) [inline c/utils.h]: Should be `a_index * b_len`?

Comment (doitian) [reply c/utils.h]: Please add a test case to cover it. For example:

```
a = 1/10
b = 2/30

a should > b
```

Comment (jjyr) [reply c/utils.h]: Fixed

**Pull request merged**